File: config/src/main/java/com/yahoo/vespa/config/protocol/CompressionType.java
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.vespa.config.protocol;

/**
 * @author Ulf Lilleengen
 */
public enum CompressionType {
    UNCOMPRESSED, LZ4;
    public static CompressionType parse(String value) {
        for (CompressionType type : CompressionType.values()) {
            if (type.name().equals(value)) {
                return type;
            }
        }
        return CompressionType.UNCOMPRESSED;
    }
}

File: config/src/vespa/config/common/compressiontype.cpp
// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
#include "compressiontype.h"

namespace config {

std::string
compressionTypeToString(const CompressionType & compressionType)
{
    switch (compressionType) {
        case CompressionType::UNCOMPRESSED:
            return "UNCOMPRESSED";
        default:
            return "LZ4";
    }
}

CompressionType
stringToCompressionType(const std::string & type)
{
    if (type.compare("UNCOMPRESSED") == 0) {
        return CompressionType::UNCOMPRESSED;
    } else {
        return CompressionType::LZ4;
    }
}

}
